content/en/blog/_posts/2021-03-03-deletes-finaliers-and-owner-references.md

```diff
@@ -0,0 +1,270 @@
+mymap-child    0      10m2s
+mymap-parent   0      10m2s
+
+$ kubectl delete configmap/mymap-parent
+configmap "mymap-parent" deleted
+$ kubectl get configmap
+No resources found in default namespace.
+```
+
+To sum things up, when there's an override owner reference from a child to a parent, deleting the parent deletes the children automatically. This is called `cascade`. The default for cascade is `true`, however, you can use the --cascade=false option for `kubectl delete` to delete an object and orphan its children. 
+
+In the following example, there is a parent and a child. Notice the owner references are still included. If I delete the parent using --cascade=false, the parent is deleted but the child still exists:
+
+```
+$ kubectl get configmap
+NAME           DATA   AGE
+mymap-child    0      13m8s
+mymap-parent   0      13m8s
+
+$ kubectl delete --cascade=false configmap/mymap-parent
+configmap "mymap-parent" deleted
+
+$ kubectl get configmap
+NAME          DATA   AGE
+mymap-child   0      13m21s
+```
+
+The --cascade option links to the propagation policy in the API, which allows you to change the order in which objects are deleted within a tree. In the following example uses API access to craft a custom delete API call with a the background propagation policy:
+
+```
+$ kubectl proxy --port=8080 &
+Starting to serve on 127.0.0.1:8080
+
+$ curl -X DELETE \
+localhost:8080/api/v1/namespaces/default/configmaps/mymap-parent \
+  -d '{
+"kind":"DeleteOptions",
+"apiVersion":"v1",
+"propagationPolicy":"Background"
+}' \
+  -H "Content-Type: application/json"
+{
+  "kind": "Status",
+  "apiVersion": "v1",
+  "metadata": {},
+  "status": "Success",
+  "details": { ... }
+```
+
+Note that the propagation policy cannot be specified on the command line using kubectl. You have to specify it using a custom API call. Simply create a proxy, so you have access to the API server from the client, and execute a `curl` command with just a URL to execute that `delete` command. 
+
+There are three different options for the propagation policy:
+
+- `Foreground`: Children are deleted before the parent (post-order)
+- `Background`: Parent is deleted before the children (pre-order)
+- `Orphan`: Owner references are ignored
+
+Keep in mind that when you delete an object and owner references have been specified, finalizers will be honored in the process. This can result in trees of objects persisting, and you end up with a partial deletion. At that point, you have to look at any existing owner references on your objects, as well as any finalizers, to understand what’s happening. 
+
+## Forcing a Deletion of a Namespace
+
+There's one situation that may require forcing finalization for a namespace. If you've deleted a namespace and you've cleaned out all of the objects under it, but the namespace still exists, use the API call `finalize`.This informs the namespace controller that it needs to remove the finalizer from the namespace and perform any cleanup:

```

Comment:
Is `finalize` an API call or a resource? It looks like a resource (you can `PUT` it) but the resource name is a verb.

Reply:
Changed - hopefully clearer.

Here's a citation to the technique: https://github.com/kubernetes/kubernetes/issues/60807#issuecomment-408599873